components/DetectAnimatedGif.coffee

```diff
@@ -22,7 +22,27 @@ exports.getComponent = ->
     async: true
     forwardGroups: true
   , (buffer, groups, out, callback) ->
-    out.send animated buffer
-    do callback
+    if Buffer.isBuffer buffer
+      out.send animated buffer
+      do callback
+      return
+    else if typeof buffer is 'string'
+      isAnimated = false
+      fs.createReadStream buffer
+        .pipe animated()
+        .once 'animated', ->
+          isAnimated = true
+          out.send true
+          do callback
+          return
+        .on 'finish', ->
+          unless isAnimated
```

Comment:
Yeah, I was thinking about it... the thing is, given a GIF as input we can have those situations:

- it's animated, so just send true
- input buffer finished, but we already sent true because it's animated, so do not send
- input buffer finished, and we didn't send true, so send false because it's not animated

If I didn't do that check it will send both true and false when we have an animated GIF.